### ISSUE TITLE
GRPO训练中间一部分后报错

### ISSUE THREAD

**Issue by winni0** [modelscope/ms-swift#3771]
**Describe the bug**
What the bug is, and how to reproduce, better with screenshots(描述bug以及复现过程，最好有截图)
训练到一部分会报错断掉
命令行：
CUDA_VISIBLE_DEVICES=4,5,6,7 \ 
NPROC_PER_NODE=4 \ 
WANDB_API_KEY=XXX \ 
swift rlhf \ 
--rlhf_type grpo \
 --model /nfs/largemodel/model/Qwen2.5-7B-Instruct \ 
--train_type lora \ 
--dataset '/nfs/largemodel/wangjuan/data/law_chinese1/DISC-Law-SFT-Pair-QA-released_alpaca.json' '/nfs/largemodel/wangjuan/data/law_chinese1/DISC-Law-SFT-Triplet-QA-released_alpaca.json' \ 
--torch_dtype bfloat16 \ 
--num_train_epochs 1 \
 --max_length 2048 \ 
--per_device_train_batch_size 1 \ 
--per_device_eval_batch_size 1 \ 
--gradient_accumulation_steps 1 \ 
--eval_steps 1000 \ --save_steps 1000 \ 
--learning_rate 1e-6 \ --save_total_limit 2 \ 
--logging_steps 5 \ 
--output_dir output \ 
--warmup_ratio 0.05 \ 
--dataloader_num_workers 4 \ 
--max_completion_length 1024 \ 
--reward_funcs format repetition \ 
--num_generations 4 \ 
--system '用户和助手之间的一段对话。用户提出一个问题，助手解决它。助手先在脑海中思考推理过程，然后再向用户提供答案。推理过程和答案分别被包裹在`<think>`和`</think>`以及`<answer>`和`</answer>`标签内，即`<think>推理过程在此</think><answer>答案在此</answer>`' \ 
--use_vllm true \ 
--vllm_gpu_memory_utilization 0.5 \
 --vllm_max_model_len 2048 \ 
--deepspeed zero3 \ 
--temperature 1.0 \ 
--top_p 1.0 \ 
--top_k 80 \ 
--log_completions true \ 
--num_infer_workers 4 \ 
--tensor_parallel_size 4 \ 
--async_generate false \ 
--report_to 'wandb' \ 
--sleep_level 1
报错：
![Image](https://github.com/user-attachments/assets/6b3cee85-c5eb-452f-907a-5a1ca311b90a)

![Image](https://github.com/user-attachments/assets/63d58e4e-fc4b-4449-a4f7-9d0b8eaec345)

![Image](https://github.com/user-attachments/assets/6068a874-94c4-4f4e-8096-00d7b90db86e)

![Image](https://github.com/user-attachments/assets/7cb5de90-31ee-4ca9-980b-076982ee9f38)

![Image](https://github.com/user-attachments/assets/969ad331-4060-4055-adb4-bb0f26542027)

![Image](https://github.com/user-attachments/assets/535c8aef-de04-4c88-87d3-383ebdf3c448)

![Image](https://github.com/user-attachments/assets/bc9433a1-3ba0-4d3c-ba9f-1b7385cf41c2)

![Image](https://github.com/user-attachments/assets/b70bd4f8-f112-4b3a-a679-435ceea77329)

![Image](https://github.com/user-attachments/assets/45d8b627-9c26-4d3b-a590-c4d1f6abcc64)

**Your hardware and system info**
Write your system info like CUDA version/system/GPU/torch version here(在这里给出硬件信息和系统信息，如CUDA版本，系统，GPU型号和torch版本等)
CUDA:11.5 H100  8*80G torch2.6.0
Package                           Version       Editable project location
--------------------------------- ------------- -------------------------
absl-py                           2.2.0
accelerate                        1.5.2
addict                            2.4.0
aiofiles                          23.2.1
aiohappyeyeballs                  2.6.1
aiohttp                           3.11.14
aiosignal                         1.3.2
airportsdata                      20250224
aliyun-python-sdk-core            2.16.0
aliyun-python-sdk-kms             2.16.5
annotated-types                   0.7.0
antlr4-python3-runtime            4.13.2
anyio                             4.9.0
astor                             0.8.1
async-timeout                     5.0.1
attrdict                          2.0.1
attrs                             25.3.0
binpacking                        1.5.2
blake3                            1.0.4
cachetools                        5.5.2
certifi                           2025.1.31
cffi                              1.17.1
charset-normalizer                3.4.1
click                             8.1.8
cloudpickle                       3.1.1
compressed-tensors                0.9.1
contourpy                         1.3.1
cpm-kernels                       1.0.11
crcmod                            1.7
cryptography                      44.0.2
cupy-cuda12x                      13.4.1
cycler                            0.12.1
dacite                            1.9.2
datasets                          3.2.0
dbus-python                       1.2.18
deepspeed                         0.14.5
depyf                             0.18.0
dill                              0.3.8
diskcache                         5.6.3
distro                            1.9.0
dnspython                         2.7.0
docker-pycreds                    0.4.0
einops                            0.8.1
email_validator                   2.2.0
exceptiongroup                    1.2.2
fastapi                           0.115.11
fastapi-cli                       0.0.7
fastrlock                         0.8.3
ffmpy                             0.5.0
filelock                          3.18.0
fonttools                         4.56.0
frozenlist                        1.5.0
fsspec                            2024.9.0
future                            1.0.0
gguf                              0.10.0
gitdb                             4.0.12
GitPython                         3.1.44
gradio                            5.22.0
gradio_client                     1.8.0
groovy                            0.1.2
grpcio                            1.71.0
h11                               0.14.0
hjson                             3.1.0
httpcore                          1.0.7
httptools                         0.6.4
httpx                             0.28.1
huggingface-hub                   0.29.3
idna                              3.10
importlib_metadata                8.6.1
iniconfig                         2.1.0
interegular                       0.3.3
jieba                             0.42.1
Jinja2                            3.1.6
jiter                             0.9.0
jmespath                          0.10.0
joblib                            1.4.2
jsonschema                        4.23.0
jsonschema-specifications         2024.10.1
kiwisolver                        1.4.8
lark                              1.2.2
latex2sympy2_extended             1.10.1
llvmlite                          0.43.0
lm-format-enforcer                0.10.11
Markdown                          3.7
markdown-it-py                    3.0.0
MarkupSafe                        3.0.2
math-verify                       0.7.0
matplotlib                        3.10.1
mdurl                             0.1.2
mistral_common                    1.5.4
modelscope                        1.24.0
mpmath                            1.3.0
ms_swift                          3.3.0.dev0    /ms-swift
msgpack                           1.1.0
msgspec                           0.19.0
multidict                         6.2.0
multiprocess                      0.70.16
nest-asyncio                      1.6.0
networkx                          3.4.2
ninja                             1.11.1.3
nltk                              3.9.1
numba                             0.60.0
numpy                             1.26.4
nvidia-cublas-cu12                12.4.5.8
nvidia-cuda-cupti-cu12            12.4.127
nvidia-cuda-nvrtc-cu12            12.4.127
nvidia-cuda-runtime-cu12          12.4.127
nvidia-cudnn-cu12                 9.1.0.70
nvidia-cufft-cu12                 11.2.1.3
nvidia-curand-cu12                10.3.5.147
nvidia-cusolver-cu12              11.6.1.9
nvidia-cusparse-cu12              12.3.1.170
nvidia-cusparselt-cu12            0.6.2
nvidia-ml-py                      12.570.86
nvidia-nccl-cu12                  2.21.5
nvidia-nvjitlink-cu12             12.4.127
nvidia-nvtx-cu12                  12.4.127
openai                            1.67.0
opencv-python-headless            4.11.0.86
orjson                            3.10.15
oss2                              2.19.1
outlines                          0.1.11
outlines_core                     0.1.26
packaging                         24.2
pandas                            2.2.3
partial-json-parser               0.2.1.1.post5
peft                              0.14.0
pillow                            11.1.0
pip                               22.0.2
platformdirs                      4.3.7
pluggy                            1.5.0
prometheus_client                 0.21.1
prometheus-fastapi-instrumentator 7.1.0
propcache                         0.3.0
protobuf                          5.29.4
psutil                            7.0.0
py-cpuinfo                        9.0.0
pyarrow                           19.0.1
pybind11                          2.13.6
pycountry                         24.6.1
pycparser                         2.22
pycryptodome                      3.22.0
pydantic                          2.10.6
pydantic_core                     2.27.2
pydub                             0.25.1
Pygments                          2.19.1
PyGObject                         3.42.1
pyparsing                         3.2.1
pytest                            8.3.5
python-dateutil                   2.9.0.post0
python-dotenv                     1.0.1
python-json-logger                3.3.0
python-multipart                  0.0.20
pytz                              2025.1
PyYAML                            6.0.2
pyzmq                             26.3.0
ray                               2.40.0
referencing                       0.36.2
regex                             2024.11.6
requests                          2.32.3
rich                              13.9.4
rich-toolkit                      0.13.2
rouge                             1.0.1
rpds-py                           0.23.1
ruff                              0.11.1
safehttpx                         0.1.6
safetensors                       0.5.3
scipy                             1.15.2
semantic-version                  2.10.0
sentencepiece                     0.2.0
sentry-sdk                        2.25.1
setproctitle                      1.3.5
setuptools                        69.5.1
shellingham                       1.5.4
simplejson                        3.20.1
six                               1.17.0
smmap                             5.0.2
sniffio                           1.3.1
sortedcontainers                  2.4.0
starlette                         0.46.1
sympy                             1.13.1
tensorboard                       2.19.0
tensorboard-data-server           0.7.2
tiktoken                          0.9.0
tokenizers                        0.21.1
tomli                             2.2.1
tomlkit                           0.13.2
torch                             2.5.1
torchaudio                        2.5.1
torchvision                       0.20.1
tqdm                              4.67.1
transformers                      4.49.0
transformers-stream-generator     0.0.5
triton                            3.1.0
trl                               0.16.0.dev0
typer                             0.15.2
typing_extensions                 4.12.2
tzdata                            2025.1
urllib3                           2.3.0
uvicorn                           0.34.0
uvloop                            0.21.0
vllm                              0.7.3
wandb                             0.19.9
watchfiles                        1.0.4
websockets                        15.0.1
Werkzeug                          3.1.3
wheel                             0.37.1
xformers                          0.0.28.post3
xgrammar                          0.1.11
xxhash                            3.5.0
yarl                              1.18.3
zipp                              3.21.0
zstandard                         0.23.0

**Additional context**
Add any other context about the problem here(在这里补充其他信息)

**Comment by hjh0119** [comment]
Feel free to reopen if you have any issues.
